# Get Player data.

In this notebook we will connect to the API to download teh data of the players tio have some condittions.

In [14]:
# The expected imports

import pandas as pd
import numpy as np

# to have access to the web service
import requests

# Define the connexion.

In [15]:
# The IP of the docker server

# Mac ip
# ip = "192.168.178.36"
# From external connexions. 
ip = "saizvirtual.ddns.net"

# Get Token

In [16]:
# URLs to connected with the API-service
url = 'http://{}:5000/token'.format(ip)

In [17]:
# Send User and Paswword
resp = requests.post(url, 
                     data={'username': 'tgarsa',
                           'password':'secret'}
                    )
# resp = requests.post(url, data={'username': 'alice', 'password':'secret'})
resp

<Response [200]>

In [18]:
token = resp.json()['access_token']
# print('Token: {}'.format(token))

token_type = resp.json()['token_type']
# print('Token Type: {}'.format(token_type))

## Define the Header

In [19]:
headers = {"Authorization": "{} {}".format(token_type, token)}

In [20]:
# headers

## Define the search 

In [21]:
search ={}
# search['id'] = 'TGS001'
# search['telefono'] = '0614297911'
# search['dni'] = '2890337'
# search['nick'] = 'smurf'
search['nick'] = 'Smurf'
# search['email'] = 'tgarsa@me.com'
# search['email'] = 'TGARSA@ME.com'
# search['pais'] = 'España'


dj_search = pd.DataFrame.from_dict(search, orient='index').transpose()
json_search = dj_search.to_json()

# Connect to the API

In [22]:
# URLs to connected with the app-service
# in this case the port is 

# url = 'http://{}:5000/get_player'.format(ip)
url = 'http://{}:5000/get'.format(ip)

In [23]:
url

'http://saizvirtual.ddns.net:5000/get'

In [24]:
resp = requests.get(url, 
                    json={'df_json': json_search, 
                          'table':'jugadores'},
                   headers=headers
                   )

In [25]:
resp

<Response [200]>

In [26]:
resp.json()

[[2,
  'TGS002',
  '2890337',
  'Tomas',
  'Garcia Saiz',
  'Smurf',
  'tgarsa@gamil.com',
  '614297912',
  'Amsterdam',
  'Holanda',
  'XXL',
  0,
  ' 02/04/24:   \n ',
  '2024-04-02T20:41:50.617902',
  '2024-04-02T20:41:50.617902'],
 [1,
  'TGS001',
  '2890340',
  'Tomas',
  'Garcia Saiz',
  'Smurf',
  'tgars2@me.com',
  '614297911',
  'Amsterdam',
  'Paises Bajos',
  'XXL',
  0,
  ' 01/04/24: Tonto el culo \n 04/04/24: Pero del toooo \n ',
  '2024-04-01T20:38:08.585569',
  '2024-04-04T11:07:30.754537']]

[[1,
  'TGS001',
  '2890340',
  'Tomas',
  'GARICA SAIZ',
  'Smurf',
  'tgars2@me.com',
  '614297914',
  'Amsterdam',
  'Paises Bajos',
  'XXL',
  'Multiples',
  'Tonto el culo',
  '2024-03-27T17:48:11.093450',
  '2024-03-27T17:48:11.093450']]
